e2e/e2etests/test_message_passing_zevm.go

```diff
@@ -0,0 +1,332 @@
+package e2etests
+
+import (
+	"fmt"
+	"math/big"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/zetacore/e2e/contracts/testdapp"
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	cctxtypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func TestMessagePassingEVMtoZEVM(r *runner.E2ERunner, args []string) {
+	if len(args) != 1 {
+		panic("TestMessagePassing requires exactly one argument for the amount.")
+	}
+	amount, ok := big.NewInt(0).SetString(args[0], 10)
+	if !ok {
+		panic("Invalid amount specified for TestMessagePassing.")
+	}
+
+	// Set destination details
+	zEVMChainID, err := r.ZEVMClient.ChainID(r.Ctx)
+	if err != nil {
+		panic(err)
+	}
+	destinationAddress := r.ZevmTestDAppAddr
+
+	// Contract call originates from EVM chain
+	auth := r.EVMAuth
```

Comment:
Just wondering why declaring a variable here instead of directly using `r.EVMAuth` at the two location?
It looks more clear to me to directly use `r.EVMAuth`

Reply:
d895bb494c3f8b659de6508e4307d4770c6d05f8